core/src/main/java/bio/terra/pearl/core/service/search/terms/LatestKitTerm.java

```diff
@@ -0,0 +1,77 @@
+package bio.terra.pearl.core.service.search.terms;
+
+import bio.terra.pearl.core.dao.kit.KitRequestDao;
+import bio.terra.pearl.core.model.kit.KitRequest;
+import bio.terra.pearl.core.service.search.EnrolleeSearchContext;
+import bio.terra.pearl.core.service.search.sql.EnrolleeSearchQueryBuilder;
+import org.jooq.Condition;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.jooq.impl.DSL.condition;
+
+/**
+ * This term fetches information about the latest kit request for an enrollee.
+ */
+public class LatestKitTerm implements SearchTerm {
+    private final String field;
+    private final KitRequestDao kitRequestDao;
+
+    public LatestKitTerm(KitRequestDao kitRequestDao, String field) {
+
+        if (!FIELDS.containsKey(field)) {
+            throw new IllegalArgumentException("Invalid field: " + field);
+        }
+
+        this.kitRequestDao = kitRequestDao;
+        this.field = field;
+    }
+
+    @Override
+    public SearchValue extract(EnrolleeSearchContext context) {
+        List<KitRequest> requests = kitRequestDao.findByEnrollee(context.getEnrollee().getId());
+        if (requests.isEmpty()) {
+            return new SearchValue();
+        }
+        requests.sort((r1, r2) -> r2.getLastUpdatedAt().compareTo(r1.getLastUpdatedAt()));
+        KitRequest latestKit = requests.get(0);
+
+        return SearchValue.ofNestedProperty(latestKit, field, FIELDS.get(field));
+    }
+
+    @Override
+    public List<EnrolleeSearchQueryBuilder.JoinClause> requiredJoinClauses() {
+        return List.of(new EnrolleeSearchQueryBuilder.JoinClause(
+                "kit_request",
+                "latest_kit",
+                "enrollee.id = latest_kit.enrollee_id"));
+    }
+
+    @Override
+    public List<EnrolleeSearchQueryBuilder.SelectClause> requiredSelectClauses() {
+        return List.of(
+                new EnrolleeSearchQueryBuilder.SelectClause("latest_kit", kitRequestDao)
+        );
+    }
+
+    @Override
```

Comment:
can you add a comment here -- I was expecting this to be done with an `order by` and `limit` but I think what you've done here is better because it's less constraining on the rest of the query.